src/utilities/device_utils.c

```diff
@@ -1174,13 +1139,31 @@ hypre_DeviceDataStream(hypre_DeviceData *data, HYPRE_Int i)
       return data->streams[i];
    }
 
-#if defined(HYPRE_USING_DEVICE_OPENMP)
-   HYPRE_CUDA_CALL(cudaStreamCreateWithFlags(&stream, cudaStreamDefault));
-#elif defined(HYPRE_USING_CUDA)
+#if defined(HYPRE_USING_CUDA)
    //HYPRE_CUDA_CALL(cudaStreamCreateWithFlags(&stream,cudaStreamNonBlocking));
    HYPRE_CUDA_CALL(cudaStreamCreateWithFlags(&stream, cudaStreamDefault));
 #elif defined(HYPRE_USING_HIP)
    HYPRE_HIP_CALL(hipStreamCreateWithFlags(&stream, hipStreamDefault));
+#elif defined(HYPRE_USING_SYCL)
+   auto sycl_asynchandler = [] (sycl::exception_list exceptions)
+   {
+      for (std::exception_ptr const& e : exceptions)
+      {
+         try
+         {
+            std::rethrow_exception(e);
+         }
+         catch (sycl::exception const& ex)
+         {
+            std::cout << "Caught asynchronous SYCL exception:" << std::endl
+                      << ex.what() << ", SYCL code: " << ex.code() << std::endl;
+         }
+      }
+   };
+
+   sycl::device* sycl_device = data->device;
+   sycl::context sycl_ctxt   = sycl::context(*sycl_device, sycl_asynchandler);
+   stream = new sycl::queue(sycl_ctxt, *sycl_device, sycl::property_list{sycl::property::queue::in_order{}});
```

Comment:
This construction of the queue object needs to happen for the default queue regardless of whether `HYPRE_USING_CUDA_STREAMS` is set.

Reply:
Maybe a simple fix is to let `HYPRE_USING_CUDA_STREAMS` defined by default with sycl

Reply:
I'm fine with that solution if it makes sense to you. I guess there is no "default stream" in sycl, since you always have to construct a queue object.